scipy/stats/stats.py

```diff
@@ -3880,6 +3881,20 @@ def ttest_rel(a, b, axis=0):
 
     """
     a, b, axis = _chk2_asarray(a, b, axis)
+
+    Ttest_relResult = namedtuple('Ttest_relResult', ('statistic', 'pvalue'))
+
+    # check both a and b
+    contains_nan, nan_policy = (_contains_nan(a, nan_policy) or
+                                _contains_nan(b, nan_policy))
+
+    if contains_nan and nan_policy == 'omit':
+        a = ma.masked_invalid(a)
+        return mstats_basic.ttest_rel(a, b, axis)
+
+    if contains_nan and nan_policy == 'propagate':
+        return Ttest_relResult(np.nan, np.nan)
```

Comment:
@aeklant these 2 lines need to be deleted (same reasoning as we just discussed for ``stats.mode``).